include/powerloader/context.hpp

```diff
@@ -40,9 +40,7 @@ namespace powerloader
 
         // Get a list of unique mirorrs if existing for the provided host name, or an empty list
         // otherwise.
-        mirror_set get_mirrors(
-            const std::string& host_name) const;  // TODO: replace std::string by std::string_view
-                                                  // as soon as a conversion is added.
+        mirror_set get_mirrors(std::string_view host_name) const;
```

Comment:
`const std::string_view&` does not work?

Reply:
To be short, this is the intended and recommended way to use `std::string_view`.

To be more complete: `std::string_view` is already a read-only reference (to a string) and the recommended way to use it pass it by value (same for `std::span` which is equivalent but not specific to strings).
But passing it by const reference just add to conversion rules, it's not more expensive (there are some platforms where there is register things at play that might be impacted but not to the point we would care).

I saw that in this codebase `string_view` was often used with `const &`, but yeah it's a bit redundant. Unfortunately it's weird at first to pass things by value this way because there is no special treatment for reference-types in the language ^^;

Reply:
Here is a summary of why (applies to all view types), there are many around but this one is more concise: https://rules.sonarsource.com/cpp/RSPEC-6009